CMakeLists.txt

```diff
@@ -34,6 +34,37 @@ endif()
 
 find_ros1_package(std_msgs REQUIRED)
 
+# find all known ROS1 message/service packages
+execute_process(
+    COMMAND "rosmsg" "list"
+    OUTPUT_VARIABLE rosmsg_output
+    ERROR_VARIABLE rosmsg_error
+)
+execute_process(
+    COMMAND "rossrv" "list"
+    OUTPUT_VARIABLE rossrv_output
+    ERROR_VARIABLE rossrv_error
+)
+if(NOT rosmsg_error STREQUAL "" OR NOT rossrv_error STREQUAL "")
+  message(FATAL_ERROR "${rosmsg_error}\n${rossrv_error}\nFailed to find ROS 1 message packages\nPlease make sure to source your ROS2 workspace BEFORE sourcing your ROS1 workspace")
```

Comment:
And the CMake code got that far in the first place the user must have sourced something from ROS 1. So maybe the message should just mention that `rosmsg` / `rossrv` failed?